src/SignalR/src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -394,7 +403,14 @@ private async Task SelectAndStartTransport(TransferFormat transferFormat)
 
             if (_transport == null)
             {
-                throw new InvalidOperationException("Unable to connect to the server with any of the available transports.");
+                if (transportExceptions.Count > 0)
+                {
+                    throw new AggregateException("Unable to connect to the server with any of the available transports.", transportExceptions);
+                }
+                else
+                {
+                    throw new InvalidOperationException("No transports are supported by the server.");
```

Comment:
Nit: In general, it should be possible to avoid InvalidOperationExceptions entirely by making you call a method correctly. In this case, a server issue could lead to this error, so maybe another exception type is appropriate.

Reply:
That's true, but what else would we throw, any suggestions?

Reply:
IOException? Or a custom exception type.